src/ol_infrastructure/providers/salt/minion.py

```diff
@@ -0,0 +1,104 @@
+"""
+Dynamic provider that uses the SaltStack API to provision a minion.
+
+Registers a minion ID with a SaltStack master and generates a keypair.  The keypair is returned so that it can be passed
+to an instance via user data to take advantage of cloud-init.
+"""
+from typing import Optional, Text
+
+from pepper import Pepper
+from pulumi import Input, Output, ResourceOptions
+from pulumi.dynamic import CreateResult, ReadResult, Resource, ResourceProvider
+from pydantic import BaseModel, SecretStr
+
+
+class OLSaltStackInputs(BaseModel):
+    minion_id: Input[Text]
+    salt_api_url: Input[Text]
+    salt_user: Input[Text]
+    salt_password: Input[SecretStr]
```

Comment:
How are we going to be managing/rotating those creds?

Reply:
That will be handled using the Pulumi stack configs, so it's out of band of Pulumi itself. I actually needed to check with you on the current status of the Salt API on QA and production.